Jennifer

layibei opened the pull request.

layibei committed:
local update for env and proxy

layibei committed:
add rag evaluation test case

layibei committed:
update test case

layibei committed:
update test case

layibei committed:
update read file

layibei committed:
update readme

layibei committed:
update html code

layibei committed:
update html

akjamie commented:
Thanks, i'm thinking we can build it as a separate repo and make it has no dependencies on the current project.

in that repo, you can only build a notebook script,
-  input - read from a file which consists of a list of Q&A.
- Call target system API and pass your loaded question, each per http call, and collect the response.
- Once done the second step, you have 2 answers/reponses for each questions, then use some python libs or LLM to do similarity check and output a score. e.g. above 90%, we can take them as the same.
- Count the correctness ratio

layibei committed:
update rag test

layibei committed:
add audit logger feature

layibei committed:
add data in audit_logger

layibei committed:
Merge branch 'main' of https://github.com/akjamie/rag-chatbot into jennifer

layibei committed:
update for add audit log

layibei committed:
update audit log

layibei committed:
update comment

layibei committed:
update for audit logger

layibei committed:
Merge branch 'main' of https://github.com/akjamie/rag-chatbot into jennifer

layibei committed:
update quick qa

layibei committed:
update quick qa

layibei committed:
Delete .idea directory

layibei committed:
Delete rules directory

layibei committed:
Update README and add QA pairs for improved documentation and knowledge base

layibei committed:
Merge branch 'jennifer' of https://github.com/layibei/rag-chatbot into jennifer

layibei committed:
Update .gitignore, refactor app.py comments, clean up requirements.txt, remove simple_audit_test.py, and adjust common_settings.py and test files for clarity and consistency

layibei committed:
Refactor get_tokenizer method in common_settings.py to simplify model loading by using a fixed model path, removing error handling for loading from HF Hub and local paths.

layibei committed:
Update model path in get_tokenizer method of common_settings.py for consistency with directory structure

layibei committed:
Add QA management routes for uploading and retrieving QA pairs

- Introduced `qa_management_routes.py` with endpoints for uploading and retrieving QA pairs.
- Implemented validation and backup functionality for uploaded QA data.
- Updated `app.py` to include the new QA management router.
- Modified `FastQAMatcher` to register instances for reloading QA data.

layibei committed:
mockup dashboard api and test

layibei committed:
move dashboard api to test folder

layibei committed:
fix the audit log async issue

layibei committed:
Enhance document retrieval process by adding parent-child document retriever support and updating .gitignore to include models directory. Refactor query processing to conditionally use the new retriever based on configuration.

layibei committed:
Enhance parent-child document retrieval functionality

layibei committed:
enhance parent document retrieval logic with improved filtering methods and error handling.

layibei committed:
Update methods to return tuples of parent IDs and sources, enhancing the retrieval process with improved filtering capabilities.

layibei committed:
update the rerank in parent-child retriever

layibei committed:
commented rerank parent doc

layibei committed:
Add mock data and query functionality for EIM issues
